source/material_model/rheology/visco_plastic.cc

```diff
@@ -240,6 +240,12 @@ namespace aspect
                   current_edot_ii = ref_strain_rate;
                 else
                   {
+                    // TODO: Use $\tau^{t+\Delta te}$ instead of $\tau^{t+\Delta t}$.
+                    // Explanation: When we compute the viscosity for the Stokes assembly
+                    // the ve stresses stored in in.composition are the rotated, averaged,
+                    // and advected stresses from the advection solve ($\tau^{t+\Delta t}$).
+                    // They should be the unaveraged, but advected stresses? How could we get
+                    // to those?
```

Comment:
Hmmm, I can't think of any easy to get the unaveraged, but advected values. However, I'm still not clear why we need to use the unaveraged values here?

Reply:
If we're doing anything at $\tau^{t+\Delta te}$ where delta te > delta t don't we have to advect into the future? It isn't clear to me how we should deal with Delta t != Delta t_e in ASPECT.